lib/Doctrine/DBAL/Schema/AbstractAsset.php

```diff
@@ -177,10 +178,12 @@ public function getName()
      */
     public function getQuotedName(AbstractPlatform $platform)
     {
-        $keywords = $platform->getReservedKeywordsList();
-        $parts    = explode('.', $this->getName());
+        $keywords     = $platform->getReservedKeywordsList();
+        $isCaseFolded = $platform->isCaseFoldedForUnquotedAssets();
+        $parts        = explode('.', $this->getName());
         foreach ($parts as $k => $v) {
-            $parts[$k] = $this->_quoted || $keywords->isKeyword($v) ? $platform->quoteIdentifier($v) : $v;
+            $caseNeedsQuoting = $isCaseFolded && preg_match('/(\p{Ll}.*\p{Lu}|\p{Lu}.*\p{Ll})/u', $v);
```

Comment:
I'd like to see this line in a separate private method. This way, the regex might not need to run when the first two matches are true in the next line.